### PR TITLE
Added special shader effects to the render library

### PR DESCRIPTION
**Pull request by mitterdoo**
I wasn't really sure how to document the color modification structure, since there doesn't seem to be any structures in the docs, so I tried to keep it brief.

### CODE REVIEW

**Comment by thegrb93** [comment]
It all lgtm

File: lua/starfall/libs_cl/render.lua [hunk 1]
```diff
@@ -748,6 +760,180 @@ function render_library.setMaterial(mat)
...
+	pp.colour:SetTexture("$fbtexture", tex)
+	surface.SetMaterial(pp.colour)
+	render.SetMaterial(pp.colour)
+
+end
+
+render_library.setMaterialEffectColourModify = render_library.setMaterialEffectColorModify
+
+
+local aspectRatio = ScrW() / ScrH() -- For some reason, render.BlurRenderTarget performs the blur based on the game's window size. This compensates for that
```
Comment (DaDamRival) [inline lua/starfall/libs_cl/render.lua]: If the user changes their resolution while in-game this could become inaccurate.

Comment (mitterdoo) [reply lua/starfall/libs_cl/render.lua]: I'll go ahead and add a fix for this, real quick

**Comment by thegrb93** [comment]
No, just calculate it in drawBlurEffect.

**Comment by mitterdoo** [comment]
> No, just calculate it in drawBlurEffect.

`drawBlurEffect` will be in the render context of a SF rendertarget, which sets the viewport to 1024x1024, overriding `ScrW()` and `ScrH()`. The values have to be fetched from a different context

**Comment by thegrb93** [comment]
I say just leave it then. User shouldn't change their resolution in-game anyway because lots of other stuff has problems when that happens too.

**Comment by thegrb93** [comment]
Isn't the original viewport dimensions stored when the rendertarget is set.

**Comment by mitterdoo** [comment]
> Isn't the original viewport dimensions stored when the rendertarget is set.

You are correct! I didn't notice that. I think I will go ahead and use that fact, instead of the hacky workaround.

File: lua/starfall/libs_cl/render.lua [hunk 2]
```diff
@@ -933,6 +927,8 @@ function render_library.drawBlurEffect(blurx, blury, passes)
 	passes = math.Clamp(blurx, 0, 100)
 
 	local rt = render.GetRenderTarget()
+	local w, h = render_library.getGameResolution()
+	local aspectRatio = ScrW() / ScrH()
```
Comment (thegrb93) [inline lua/starfall/libs_cl/render.lua]: Should be w / h right? Also I would avoid using render_library function since users will have control over that.

Comment (mitterdoo) [reply lua/starfall/libs_cl/render.lua]: Got it. Should be fixed now

**Pull request merged**